### ISSUE TITLE
understanding the basics of dataloader and procedure

### ISSUE THREAD

**Issue by seyeeet** [Sha-Lab/FEAT#40]
Thanks for sharing your code, it is so useful and great.
I'm new to FSL, I think I understand most of the stuff about it and I know what is query and support.
I am not able to understand the concept of the query when it is more than `1`? can you please clear that concept for me? what does it mean to have query `15`?

to be more specific, for `way=5`, `shot=1`, and `query=15`, the data is in the shape of `80x3xwxh`.
I don't understand this. does `instance` define as `query+support`? if so, I guess for each instance (query+suppor) we have (5 support + 1 query), am I correct? and then `15*6` should be `90`? can you please explain what I'm missing here, where the `80` comes from?

In this case (input = `80x3xwxh`), the outputs of model is `logits` and `reg_logits` which are in the shape of `75x5` and `80x5`. would you also explain what those dimensions represent? 
I understand that 5 is the number of ways (classes in this episode), but cannot figure out the rest

**Comment by Han-Jia** [comment]
Here we train a model based on the support set, and the model should make predictions over query set instances. Both support and query sets have the same number of ways (number of classes) in our implementation. 

In the support set, there are way=5 classes and shot=1 instance per class. In the query set, there are way=5 classes and query=15 instances per class. Therefore, in each batch, we have 5*1 + 5*15 =80 instances, and then we split them into support and query sets. Finally, the model makes predictions on the 75 query instances.

BTW, the reg_logits  are computed on the whole batch, so its output has size 80.

**Comment by seyeeet** [comment]
Thank you for your response.
I see, got it.
 so `logits` represent the prediction on the query.
Regarding the `reg_logits` does it mean that in addition to the FSL we also do classification on the whole batch (80 images)?

**Comment by Han-Jia** [comment]
yes. 

reg_logits is the logits for the contrastive regularizer,  which is only used during meta-training. Thus we apply it over all the instances in the batch (both support and query sets).

**Comment by seyeeet** [comment]
Thank you Han-Jia for the clarifications!
